cmd/apprepository-controller/controller_test.go

```diff
@@ -344,14 +350,106 @@ func Test_newSyncJob(t *testing.T) {
 												SecretKeyRef: &corev1.SecretKeySelector{LocalObjectReference: corev1.LocalObjectReference{Name: "apprepo-my-charts-secrets"}, Key: "AuthorizationHeader"}},
 										},
 									},
+									VolumeMounts: []corev1.VolumeMount{},
 								},
 							},
+							Volumes: []corev1.Volume{},
 						},
 					},
 				},
 			},
 			"kubeapps/v2.3",
 		},
+		{
+			"my-charts",
+			&apprepov1alpha1.AppRepository{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "AppRepository",
+					APIVersion: "kubeapps.com/v1alpha1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-charts",
+					Namespace: "kubeapps",
+					Labels: map[string]string{
+						"name":       "my-charts",
+						"created-by": "kubeapps",
+					},
+				},
+				Spec: apprepov1alpha1.AppRepositorySpec{
+					Type: "helm",
+					URL:  "https://charts.acme.com/my-charts",
+					Auth: apprepov1alpha1.AppRepositoryAuth{
```

Comment:
should we add a test for the customCA and auth header? just for completeness sake?